### PR TITLE
Feature/cancel revoke registration

### PR DESCRIPTION
**Pull request by JohnJAMoore**
Add Delete registration endpoint

Issue: Scottish-Natural-Heritage/Licensing#586

### CODE REVIEW

File: util/db/migrations/20210601161145-delete-registration.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+'use strict';
```
Comment (MikeCoats) [inline util/db/migrations/20210601161145-delete-registration.js:1]: Could you rename this file to better describe the action the migration takes? The filename is saved in the Sequelize Meta table, so a good descriptive name is essential.

It doesn't 'delete a registration' but really 'creates the revocation table', so `util/db/migrations/20210601161145-create-revocation.js` would be a better name for it.

File: src/router.js [hunk 2]
```diff
@@ -134,6 +135,69 @@ router.put('/registrations/:id', async (request, response) => {
...
+      return response.status(500).send({message: `Could not update revocation ${newId}.`});
+    }
+
+    // Soft delete the registration that is being revoked.
+    const deleteRegistration = await Registration.delete(existingId);
+
+    // If they're not successful, send a 500 error.
+    if (deleteRegistration === undefined) {
+      return response.status(500).send({message: `Could not delete Registration ${existingId}.`});
+    }
```
Comment (MikeCoats) [inline src/router.js:191]: Shouldn't these three operations be part of a transaction?

It looks like Sequelize makes this quite easy.
https://sequelize.org/v5/manual/transactions.html#managed-transaction--auto-callback-

**Review by MikeCoats — CHANGES_REQUESTED** [review]
Two small changes, but apart from that it's fine!

File: src/router.js (hunk 2 above)
Comment (JohnJAMoore) [reply src/router.js]: resolved in https://github.com/Scottish-Natural-Heritage/trap-registration-api/pull/25/commits/99ddcdba117507c42a1f9d5f11f89736b78a6fa2

File: util/db/migrations/20210601161145-delete-registration.js (hunk 1 above)
Comment (JohnJAMoore) [reply util/db/migrations/20210601161145-delete-registration.js]: resolved in https://github.com/Scottish-Natural-Heritage/trap-registration-api/pull/25/commits/ec4ffc241448f099d9575dbc2e68a8bd207a7d8b

**Review by paulcoxsnh — APPROVED** [review]
Looks good :)

**Pull request merged**